src/DGtal/geometry/volumes/BoundedLatticePolytope.h

```diff
@@ -0,0 +1,878 @@
+       
+       @param hs any half-space constraint.
+       @param large tells if the inequality is large (true) or strict (false).
+       @param valid_edge_constraint when 'true', tells that there are
+       half-spaces that represents th constraints on edges (n-2 cells)
+       lying between two faces (n-1 cells) pointing to different
+       orthants.
+       @return the index of the constraint in the polytope.
+
+       @note For now complexity is O(n) where n=A.rows() because it
+       checks if a parallel closed half-space defines already a face
+       of the polytope.
+     */
+    unsigned int cut( const HalfSpace & hs, bool large = true,
+		      bool valid_edge_constraint = false );
+    
+    /**
+       Swaps the content of this object with other. O(1) complexity.
+       @param other any other BoundedLatticePolytope.
+    */
+    void swap( BoundedLatticePolytope & other );
+
+
+    /**
+     * Dilates this polytope P by t.
+     * @param t any integer.
+     * @return a reference to 'this', which is now the polytope tP.
+     */
+    Self& operator*=( Integer t );
+
+    /**
+     * Minkowski sum of this polytope with a unit segment aligned with some axis.
+     *
+     * @param s any unit segment.
+     * @return a reference to 'this'.
+     */
+    Self& operator+=( UnitSegment s );
+
+    /**
+     * Minkowski sum of this polytope with an axis-aligned unit cell.
+     *
+     * @param c any unit cell.
+     * @return a reference to 'this'.
+     */
+    Self& operator+=( UnitCell c );
+
+    /**
+     * Minkowski sum of this polytope with a strict unit segment aligned with some axis.
+     *
+     * @param s any strict unit segment.
+     * @return a reference to 'this'.
+     */
+    Self& operator+=( RightStrictUnitSegment s );
+
+    /**
+     * Minkowski sum of this polytope with an axis-aligned strict unit cell.
+     *
+     * @param c any strict unit cell.
+     * @return a reference to 'this'.
+     */
+    Self& operator+=( RightStrictUnitCell c );
+
+    /**
+     * Minkowski sum of this polytope with a strict unit segment aligned with some axis.
+     *
+     * @param s any strict unit segment.
+     * @return a reference to 'this'.
+     */
+    Self& operator+=( LeftStrictUnitSegment s );
+
+    /**
+     * Minkowski sum of this polytope with an axis-aligned strict unit cell.
+     *
+     * @param c any strict unit cell.
+     * @return a reference to 'this'.
+     */
+    Self& operator+=( LeftStrictUnitCell c );
+    /// @}
+
+    // ----------------------- Enumeration services ------------------------------
+  public:
+
+    /// @name Enumeration services (counting, get points in polytope)
+    /// @{
+    
+    /**
+     * Computes the number of integer points lying within the polytope.
+     *
+     * @return the number of integer points lying within the polytope.
+     *
+     * @note Quite slow: obtained by checking every point of the polytope domain.
```

Comment:
More info? Like "(number of integer points of the bounding box times the isInterior() cost)" ?